doc/source/cross-language.rst

```diff
@@ -0,0 +1,246 @@
+Cross Language
+==============
+
+This page will cover some more cross language examples of using Ray's flexible programming model.
+
+Python calling Java
+-------------------
+
+You can write a Java class as follows:
+
+.. code-block:: java
+
+  // A regular Java class.
+  public class Counter {
+
+    private int value = 0;
+
+    public int increment() {
+      this.value += 1;
+      return this.value;
+    }
+
+    public static int add(int a, int b) {
+      return a + b;
+    }
+  }
+
+Then you can create the Java actor from Python, or call Java remote function
+from Python:
+
+.. code-block:: python
+
+  import ray
+
+  ray.init(_include_java=True, _load_code_from_local=True)
+
+  # Define a Java class.
+  counter_class = ray.java_actor_class(
+        "<your java package>.Counter")
+
+  # Create a Java actor and call actor method.
+  counter = counter_class.remote()
+  obj_ref1 = counter.increment.remote()
+  assert ray.get(obj_ref1) == 1
+  obj_ref2 = counter.increment.remote()
+  assert ray.get(obj_ref2) == 2
+
+  # Define a Java function.
+  add_function = ray.java_function(
+        "<your java package>.Counter", "add")
+  
+  # Call the Java remote function.
+  obj_ref3 = add_function.remote(1, 2)
+  assert ray.get(obj_ref3) == 3
+
+  ray.shutdown()
+  
+Java calling Python
+-------------------
+
+You can write a Python module as follows:
+
+.. code-block:: python
+
+  class Counter(object):
+    def __init__(self):
+        self.value = 0
+
+    def increment(self):
+        self.value += 1
+        return self.value
+
+
+  def add(a, b):
+      return a + b
+
+Then you can create Python actor from Java, or call Python remote function
+from Java:
+
+.. code-block:: java
+
+  import io.ray.api.Ray;
+  import io.ray.api.function.PyActorClass;
+  import io.ray.api.function.PyActorMethod;
+  import io.ray.api.function.PyFunction;
+  import org.testng.Assert;
+
+  public class MyRayApp {
+
+    public static void main(String[] args) {
+      Ray.init();
+      
+      // Define a Python class.
+      PyActorClass actorClass = PyActorClass.of(
+          "<your python module>", "Counter");
```

Comment:
also, use real python module here. so this example will be easier to understand.

Reply:
OK.